### ISSUE TITLE
How to use react-mde with typescript

### ISSUE THREAD

**Issue by ihaichao** [andrerpena/react-mde#53]
````
import React from 'react'
const ReactMde = require('react-mde')
const { ReactMdeCommands } = require('react-mde') 

interface ComponentProps {}

interface ComponentState {
	reactMdeValue: object
}

class WriteArticle extends React.Component<ComponentProps, ComponentState> {
	constructor(props: object) {
		super(props)
		this.state = {
			reactMdeValue: {
				text: '',
				selection: null
			}
		}
	}

	handleValueChange = (value: object) => {
		this.setState({reactMdeValue: value})
	}

	render() {
		return (
			<div className="container">
				<ReactMde
					textAreaProps={{
							id: 'ta1',
							name: 'ta1',
					}}
					value={this.state.reactMdeValue}
					onChange={this.handleValueChange}
					commands={ReactMdeCommands}
				/>
			</div>
		)
	}
}

export default WriteArticle
````
The above code runs with error as below:
`warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: object.`

react: 16.1.0
react-mde: 1.0.3

**Comment by daanishnasir** [comment]
``
import React, { Component } from 'react';
import ReactMde, { ReactMdeCommands } from 'react-mde';     //needs to be direct source to file 
import Showdown from "showdown";


import 'react-mde/lib/styles/css/react-mde.css';
import 'react-mde/lib/styles/css/react-mde-command-styles.css';
import 'react-mde/lib/styles/css/markdown-default-theme.css';



export default class MarkdownEditor extends Component {

    converter: Showdown.Converter;

    constructor(props) {
        super(props);
        
        this.state = {
            reactMdeValue: {
                text: '', 
                selection: null},
            HTML: ""
        };

        this.converter = new Showdown.Converter();
    }


    handleValueChange = (value) => {
       

        let newHTML = this.converter.makeHtml(value.text) || '<p>&nbsp</p>';

        this.setState({
            reactMdeValue: value,
            HTML: newHTML
        });


    }


    render() {


        return (
            <div className="container">
                <ReactMde
                    textAreaProps={{                  
                        id: 'ta1',
                        name: 'ta1',
                    }}
                    value={this.state.reactMdeValue}
                    onChange={this.handleValueChange}
                    commands={ReactMdeCommands}
                />
            </div>
        );
    }
}
My code right now and it works

**Comment by ihaichao** [comment]
@daanishnasir It works in javascript, but doesn't work in typescript.

**Comment by andrerpena** [comment]
I'm sorry. My fault.

I'm compiling the TS(X) files to JavaScript and I'm indeed generating type definitions (.d.ts) files, but if you couldn't make it to work, I certainly got something wrong.

I made a demo to test the NPM package, for JS here: https://github.com/andrerpena/react-mde-js-demo

I'm going to do the exact same demo for TypeScript and I'm making this a priority now. I'll update this thread as soon as I get anything.

**Comment by ihaichao** [comment]
@andrerpena Really appreciate that.

**Comment by andrerpena** [comment]
Fixed on 1.0.6

**Comment by andrerpena** [comment]
The demo is here, BTW: https://github.com/andrerpena/react-mde-ts-demo
